### PR TITLE
fix: mm trade steps & lock

### PR DESCRIPTION
**Pull request by ChefJerry**
---

## PR-Codex overview
This PR focuses on enhancing swap functionality and improving translations. 

### Detailed summary
- Added a new type `MMCommitTrade`
- Updated `usePermit2Requires.ts` and `usePermit2.ts`
- Added `useThrottleFn` hook
- Modified components related to swap transactions
- Updated translations in `translations.json`

> The following files were skipped due to too many changes: `apps/web/src/views/Swap/V3Swap/hooks/useAllTypeBestTrade.ts`, `apps/web/src/views/Swap/MMLinkPools/hooks/useSwapCallback.ts`, `apps/web/src/hooks/useApproveCallback.ts`, `apps/web/src/views/Swap/MMLinkPools/hooks/useApprove.ts`, `apps/web/src/views/Swap/V3Swap/containers/SwapCommitButtonV2.tsx`, `apps/web/src/views/Swap/MMLinkPools/components/MMCommitButtonV2.tsx`, `apps/web/src/views/Swap/V3Swap/index.tsx`, `apps/web/src/views/Swap/V3Swap/hooks/useConfirmModalStateV2.ts`, `apps/web/src/views/Swap/MMLinkPools/hooks/useMMConfirmModalState.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/web/src/hooks/useApproveCallback.ts [hunk 1]
```diff
@@ -186,6 +192,13 @@ export function useApproveCallback(
     ],
   )
 
+  const approveNoCheck = useCallback(
+    async (overrideAmountApprove?: bigint) => {
+      return approve(overrideAmountApprove, false)
+    },
+    [approve],
+  )
```
Comment (chefjackson) [inline apps/web/src/hooks/useApproveCallback.ts:200]: hmm it's getting worse adding these bypass functions

Comment (chefjackson) [reply apps/web/src/hooks/useApproveCallback.ts:200]: if without state check, what if the user double clicked on the approve button? Does it trigger twice the approve tx?

Comment (izayl) [reply apps/web/src/hooks/useApproveCallback.ts:200]: but it depends use case, on current universal router case, it triggered by hooks automaticly

Comment (ChefJerry) [reply apps/web/src/hooks/useApproveCallback.ts:200]: it's depends on use case, on universal router case, it only triggered by hooks, not triggered by user event

File: apps/web/src/views/Swap/V3Swap/index.tsx [hunk 2]
```diff
@@ -35,42 +55,40 @@ export function V3SwapForm() {
...
     if (featureEnabled) {
-      return mm?.isMMBetter ? (
-        <MMCommitButtonV2 {...mm} />
+      return mmCurrentTrade?.isMMBetter ? (
+        <MMCommitButtonV2 {...mmCurrentTrade} setLock={setLock} />
       ) : (
-        <SwapCommitButtonV2 trade={trade} tradeError={error} tradeLoading={!tradeLoaded} />
+        <SwapCommitButtonV2 trade={ammCurrentTrade} tradeError={error} tradeLoading={!tradeLoaded} setLock={setLock} />
       )
     }
```
Comment (chefjackson) [inline apps/web/src/views/Swap/V3Swap/index.tsx:73]: Shall we have a separate `useBestTrade` hook to include the compare logic? And I think we can make use of the `enabled` field from `useQuery` to pause the rfq if we're in the middle of the operation
```typescript
const [isQuotingPaused, setIsQuotingPaused] = useState(false)
const trade = useBestTrade(isQuotingPaused)

<CommitButton
  trade={trade}
  onSwap={pause}
  onFinished={unpause}
/>
``` 

Does it make sense?

File: apps/web/src/utils/client.ts [hunk 3]
```diff
@@ -37,3 +37,32 @@ export const { publicClient, chains } = configureChains(
...
+  {
+    batch: {
+      multicall: {
+        batchSize: 1024 * 200,
+        wait: 16,
+      },
+    },
+    pollingInterval: 6_000,
+  },
+)
```
Comment (chefjackson) [inline apps/web/src/utils/client.ts:68]: So the issue we talked about also occurs when we use the same rpc endpoint right? If so there's no need for dedicated endpoint? Shall we just remove this part?

File: apps/web/src/views/Swap/MMLinkPools/components/MMCommitButtonV2.tsx [hunk 4]
```diff
@@ -63,6 +61,7 @@ export function MMSwapCommitButtonV2({
   currencyBalances,
   recipient,
   onUserInput,
+  setLock,
```
Comment (chefjackson) [inline apps/web/src/views/Swap/MMLinkPools/components/MMCommitButtonV2.tsx:64]: I'm going to revisit my previous comment here https://github.com/pancakeswap/pancake-frontend/pull/9274/#discussion_r1520802664

It's not quite readable to me to have a `setLock` prop on a button. Can we redesign the interface?

File: apps/web/src/utils/client.ts (hunk 3 above)
Comment (ChefJerry) [reply apps/web/src/utils/client.ts:68]: sure, have tested, even same rpc will have same issue, let me remove these loc

File: apps/web/src/views/Swap/V3Swap/hooks/useAllTypeBestTrade.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  }, [])
+
+  const resumeQuoting = useCallback(() => {
+    setIsQuotingPaused(false)
+  }, [])
+
+  const refreshTrade = useCallback(() => {
+    throttle(() => {
+      refresh()
+    }, 3000)
```
Comment (chefjackson) [inline apps/web/src/views/Swap/V3Swap/hooks/useAllTypeBestTrade.ts:42]: missing return here? `throttle` should be returning a new function

File: apps/web/src/views/Swap/V3Swap/hooks/useConfirmModalStateV2.ts [hunk 6]
```diff
@@ -162,16 +220,17 @@ const useConfirmActions = (
...
-              confirmations: chainId === mainnet.id ? 2 : 1,
-            })
+            await retryWaitForTransaction({ hash: result.hash })
           }
           let newAllowanceRaw: bigint = amountToApprove?.quotient ?? 0n
           // check if user really approved the amount trade needs
           try {
-            const { data } = await refetch()
-            newAllowanceRaw = data ?? 0n
+            // const { data } = await refetch()
```
Comment (chefjackson) [inline apps/web/src/views/Swap/V3Swap/hooks/useConfirmModalStateV2.ts:228]: better remove unused code

**Review by chefjackson — APPROVED** [review]

**Pull request merged**